log4j-redis/src/main/java/org/apache/logging/log4j/redis/appender/RedisManager.java

```diff
@@ -0,0 +1,103 @@
+package org.apache.logging.log4j.redis.appender;
+
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.appender.AbstractManager;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.exceptions.JedisConnectionException;
+
+import java.nio.charset.Charset;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+class RedisManager extends AbstractManager {
+
+    private final byte[][] byteKeys;
+    private final String host;
+    private final int port;
+    private final Charset charset;
+    private JedisPool jedisPool;
+
+    RedisManager(LoggerContext loggerContext, String name, String[] keys, String host, int port, Charset charset) {
+        super(loggerContext, name);
+        this.byteKeys = new byte[keys.length][];
+        for (int i = 0; i < keys.length; i++) {
+            this.byteKeys[i] = keys[i].getBytes(charset);
+        }
+        this.charset = charset;
+        this.host = host;
+        this.port = port;
+    }
+
+    JedisPool createPool(String host, int port, boolean ssl) {
+        JedisPoolConfig poolConfig = new JedisPoolConfig();
+        poolConfig.setMaxIdle(5);
+        poolConfig.setMinIdle(1);
+        poolConfig.setTestOnBorrow(true);
+        poolConfig.setTestOnReturn(true);
+        poolConfig.setTestWhileIdle(true);
+        poolConfig.setNumTestsPerEvictionRun(10);
+        poolConfig.setTimeBetweenEvictionRunsMillis(60000);
+        return new JedisPool(poolConfig, host, port, ssl);
+    }
+
+    public void startup() {
+        jedisPool = createPool(host, port, false);
+    }
+
+    public void send(byte[] value) {
+        runInPool(jedis -> {
+            for (byte[] key: byteKeys) {
+                jedis.rpush(key, value);
+            }
+            return null;
+        }, ex -> {
+            LOGGER.error("Unable to send audit log message to ddd. Ensure that it's running on {} : {}", host, port, ex);
+            return null;
+        });
```

Comment:
Avoid lambda allocation. 2.x mostly depends on java 7, building against that version may help avoid these

Reply:
No other values are used for `sendFunction` and `onError`, can we avoid creating and passing functions?